src/Sarif/Autogenerated/ReportingDescriptorEqualityComparer.cs

```diff
@@ -235,10 +235,7 @@ public int GetHashCode(ReportingDescriptor obj)
                     foreach (var value_5 in obj.DeprecatedGuids)
                     {
                         result = result * 31;
-                        if (value_5 != null)
```

Comment:
>null [](http://example.com/codeflow?start=39&length=4)

Wouldn't the right test here be 

if (value_5 != default)  
  OR
if (value_5 != Guid.Empty)  

or do we require/benefit from processing the hashcode for these empty items?

Note that an empty guid is not a real guid. OTOH, we should regard two lists as unique if the only difference between them is that one contains Guid.Empty and the other does not, right?

So maybe there is no change here...

Reply:
I thought about this a well, 
this change is auto-generated not hand made, I believe this is correct because deprecatedGuids  is a list of *valid* GUID and `null` is not a valid GUID so it can not be in the list, so each item in the list is not nullable and then the null check is not needed. the DeprecatedGuids is defined as C# List<Guid>, let me know,
the definistion says "an array of zero or more unique (§3.7.3) GUID-valued strings (§3.5.3)":
A reportingDescriptor object MAY contain a property named deprecatedGuids whose value is an array of zero or more unique (§3.7.3) GUID-valued strings (§3.5.3) each of which was used by a previous version of the tool as the value of the guid property (§3.49.5) for this object.